zookeeper-recipes/zookeeper-recipes-queue/src/main/java/org/apache/zookeeper/recipes/queue/DistributedQueue.java

```diff
@@ -94,13 +98,17 @@ private String smallestChildName() throws KeeperException, InterruptedException
         long minId = Long.MAX_VALUE;
         String minName = "";
 
-        List<String> childNames;
+        List<String> childNames = null;
 
         try {
             childNames = zookeeper.getChildren(dir, false);
         } catch (KeeperException.NoNodeException e) {
             LOG.warn("Unexpected exception", e);
             return null;
+        } finally {
+            if (childNames == null) {
+                return null;
```

Comment:
returning in a finally block is a code smell.
can we return "null" when we are out of the try block ?

Reply:
Ok, I will make it outside try block.